frontend/ui/data/keyboardlayouts/zh_ime.lua

```diff
@@ -0,0 +1,346 @@
+-----------------------------------------
+-- General Chinese input method engine --
+-----------------------------------------
+local logger = require("logger")
+local util = require("util")
+
+local function binarysearch( tbl,value,fcompval,reversed )
```

Comment:
`varname, comma, space, varname, comma, space`
(unless this is cut & pasted from some other code, and you don't want to introduce diff for later update/sync)